primefaces-integration-tests/src/test/java/org/primefaces/integrationtests/tree/Tree001Test.java

```diff
@@ -78,13 +80,64 @@ public void testBasicAndToggling(Page page) {
         assertConfiguration(tree.getWidgetConfiguration());
     }
 
+    @Test
+    @Order(2)
+    @DisplayName("Tree: Ajax update")
+    public void testAjaxUpdate(Page page) {
+        // Arrange
+        Tree tree = page.tree;
+        Assertions.assertNotNull(tree);
+
+        List<TreeNode> children = tree.getChildren();
+
+        Assertions.assertNotNull(children);
+        Assertions.assertEquals(3, children.size());
+
+        TreeNode first = children.get(0);
+        Assertions.assertEquals("Documents", first.getLabelText());
+
+        TreeNode second = children.get(1);
+        Assertions.assertEquals("Events", second.getLabelText());
+
+        TreeNode third = children.get(2);
+        Assertions.assertEquals("Movies", third.getLabelText());
+
+        // Act
+        PrimeSelenium.guardAjax(page.buttonUpdate).click();
```

Comment:
Please expand (toggle) one node before clicking update-button. So we can check this node is still expanded after update.

Reply:
Added.